query-engine/query-engine-napi/src/lib.rs

```diff
@@ -109,16 +112,44 @@ fn version(ctx: CallContext) -> napi::Result<JsUnknown> {
     ctx.env.to_js_value(&version)
 }
 
-#[js_function(0)]
-fn dmmf(ctx: CallContext) -> napi::Result<JsObject> {
-    let this: JsObject = ctx.this_unchecked();
-    let engine: &QueryEngine = ctx.env.unwrap(&this)?;
-    let engine: QueryEngine = engine.clone();
+#[js_function(1)]
+fn dmmf(ctx: CallContext) -> napi::Result<JsUnknown> {
+    #[derive(serde::Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    struct GetConfigOptions {
+        datamodel: String,
+        #[serde(default)]
+        enable_raw_queries: bool,
+    }
 
-    ctx.env
-        .execute_tokio_future(async move { Ok(engine.dmmf().await?) }, |&mut env, dmmf| {
-            env.create_string(&serde_json::to_string(&dmmf).unwrap())
-        })
+    let options = ctx.get::<JsUnknown>(0)?;
+    let options: GetConfigOptions = ctx.env.from_js_value(options)?;
+
+    let GetConfigOptions {
+        datamodel,
+        enable_raw_queries,
+    } = options;
+
+    let template = DatamodelConverter::convert_string(datamodel.clone());
+    let config =
+        datamodel::parse_configuration(&datamodel).map_err(|errors| ApiError::conversion(errors, &datamodel))?;
+    let capabilities = match config.subject.datasources.first() {
+        Some(datasource) => datasource.capabilities(),
+        None => ConnectorCapabilities::empty(),
+    };
+    // temporary code duplication
+    let internal_data_model = template.build("".into());
```

Comment:
we do need to pass the database name here, so we should change the `exec_loader.rs` so it has a separate function to fetch the info.